File: spec/factories/transactions.rb
FactoryBot.define do
  factory :transaction do
    user { nil }
    from_currency { "MyString" }
    to_currency { "MyString" }
    amount_from { "9.99" }
    amount_to { "9.99" }
    rate { "9.99" }
    status { "MyString" }
    error_message { "MyString" }
  end
end

File: frontend/src/hooks/types/transactions.ts
export interface Transaction {
  uuid: string;
  from_currency: string;
  to_currency: string;
  amount_from: string;
  amount_to: string;
  rate: string;
  status: string;
  error_message: string | null;
  created_at: string;
}

export interface TransactionsResponse {
  transactions: Transaction[];
  page: number;
  limit: number;
  total: number;
  total_pages: number;
  has_next: boolean;
  has_prev: boolean;
}
